modules/core/SolidSphere.cpp

```diff
@@ -0,0 +1,42 @@
+#include "SolidSphere.h"
+#include "PrimTriangle.h"
+
+
+rt::CSolidSphere::CSolidSphere(rt::ptr_shader_t pShader, const Vec3f &origin, float radius, size_t sides) : CSolid(origin)
+{
+    m_origin = origin;
+    m_radius = radius;
+    m_sides = sides;
+
+    // This is quite straight forwards - The function creates 4 points and creates a quad from them
+    // It uses UV coordinates to so. Except at the cap where it creates triangles from the "north/south poles"
+    // to the points from a layer below.
+    for (int i = 0; i <= sides; i++)
+    {
+        for (int j = 0; j < sides; j++)
+        {
+            Vec3f a, b, c, d, n_a, n_b, n_c, n_d;;
+            Vec2f t_a, t_b, t_c, t_d;
+            computeCoords(i, j, a, t_a, n_a);
+            computeCoords(i+1, j, b, t_b, n_b);
+            computeCoords(i, j+1, c, t_c, n_c);
+            computeCoords(i + 1, j + 1, d, t_d, n_d);
+            add(std::make_shared<CSolidQuad>(pShader, a, b, d, c, t_a, t_b, t_d, t_c, n_a, n_b, n_d, n_c));
```

Comment:
The top and bottom parts should be modeled with triangles

Reply:
**Top View**:

![image](https://user-images.githubusercontent.com/41652846/100075564-660f5e80-2e40-11eb-8625-36e52c4806e9.png)

**Bottom View**:

![image](https://user-images.githubusercontent.com/41652846/100075575-6a3b7c00-2e40-11eb-9fc3-2cabdc697693.png)